Sources/FoundationEssentials/Data/Data+Writing.swift

```diff
@@ -314,15 +325,26 @@ internal func writeToFile(path inPath: PathOrURL, data: Data, options: Data.Writ
 }
 
 internal func writeToFile(path inPath: PathOrURL, buffer: UnsafeRawBufferPointer, options: Data.WritingOptions, attributes: [String : Data] = [:], reportProgress: Bool = false) throws {
+#if os(WASI) // `.atomic` is unavailable on WASI
+    try writeToFileNoAux(path: inPath, buffer: buffer, options: options, attributes: attributes, reportProgress: reportProgress)
```

Comment:
Should this also throw like the public entry point, in case someone adds a new call to it that circumvents the outer one?

Reply:
Or, put it inside the helper function `writeToFileAux` instead and leave this one without a diff?

Reply:
@parkera Thank you for reviewing. I tried my best to understand your comments, but I may not be able to imagine the actual examples you are concerned about.

> Should this also throw like the public entry point, in case someone adds a new call to it that circumvents the outer one?

I don't think such a case will happen, or we can notice it immediately if it happens.

If someone adds a new call like `try writeToFile(path: ..., buffer: ..., options: .atomic)`, then it will cause a compile error when compiling swift-foundation for WASI, because `.atomic` is unavailable if `os(WASI)`.

Example:

```
/home/kebo/swift-foundation/Sources/FoundationEssentials/Data/Data+Writing.swift:329:69: error: 'atomic' is unavailable: atomic writing is unavailable in WASI because temporary files are not supported
327 | func foo() throws {
328 |     try Data().withUnsafeBytes { buf in
329 |         try writeToFile(path: .path("/foo"), buffer: buf, options: .atomic)
    |                                                                     `- error: 'atomic' is unavailable: atomic writing is unavailable in WASI because temporary files are not supported
330 |     }
331 | }

/home/kebo/swift-foundation/Sources/FoundationEssentials/Data/Data.swift:2063:27: note: 'atomic' has been explicitly marked unavailable here
2061 |         @available(*, unavailable, message: "atomic writing is unavailable in WASI because temporary files are not supported")
2062 | #endif
2063 |         public static let atomic = WritingOptions(rawValue: 1 << 0)
     |                           `- note: 'atomic' has been explicitly marked unavailable here
2064 |         
2065 |         /// An option that attempts to write data to a file and fails with an error if the destination file already exists.
```

> Or, put it inside the helper function writeToFileAux instead and leave this one without a diff?

Since `.atomic` and `writeToFileAux` are unavailable in WASI, it is impossible to leave this method without a diff.

```
/home/kebo/swift-foundation/Sources/FoundationEssentials/Data/Data+Writing.swift:328:26: error: 'atomic' is unavailable: atomic writing is unavailable in WASI because temporary files are not supported
326 | 
327 | internal func writeToFile(path inPath: PathOrURL, buffer: UnsafeRawBufferPointer, options: Data.WritingOptions, attributes: [String : Data] = [:], reportProgress: Bool = false) throws {
328 |     if options.contains(.atomic) {
    |                          `- error: 'atomic' is unavailable: atomic writing is unavailable in WASI because temporary files are not supported
329 |         try writeToFileAux(path: inPath, buffer: buffer, options: options, attributes: attributes, reportProgress: reportProgress)
330 |     } else {
```